File: UniversalBre/token.cpp
#include "token.h"

core::token::token(token_type type) : _token_type(type)
{
}

core::token::token(token_type type, std::wstring lexeme)
    : _token_type(type), _lexeme(lexeme)
{

}

core::token::~token()
{
}

core::token_type core::token::get_type()
{
    return _token_type;
}

std::wstring core::token::get_lexeme()
{
    return _lexeme;
}

File: RustCode/src/token.rs
use token_type::TokenType;

#[derive(Debug, Clone)]
pub struct Token {
	line: usize,
	column: usize,
	token_type: TokenType,
	lexeme: String
}

impl Token {
	pub fn init(line: usize, column: usize, token_type: TokenType, lexeme: String) -> Token {
		Token {
            line: line,
            column: column,
            token_type: token_type,
            lexeme: lexeme
        }
	}

	pub fn get_token_type(&self) -> &TokenType {
		&self.token_type
	}

	pub fn get_lexeme(&self) -> &String {
		&self.lexeme
	}

	pub fn get_line(&self) -> usize {
		self.line
	}

	pub fn get_column(&self) -> usize {
		self.column
	}
}